Source/VMware.vSphereDSC/Classes/VMHostBaseDSC.ps1

```diff
@@ -38,4 +50,97 @@ class VMHostBaseDSC : BaseDSC {
             throw "VMHost with name $($this.Name) was not found. For more information: $($_.Exception.Message)"
         }
     }
+
+    <#
+    .DESCRIPTION
+
+    Checks if the specified VMHost is in Maintenance mode and if not, throws an exception.
+    #>
+    [void] EnsureVMHostIsInMaintenanceMode($vmHost) {
+        if ($vmHost.ConnectionState.ToString() -ne $this.MaintenanceState) {
+            throw "The Resource Update operation requires the VMHost $($vmHost.Name) to be in a Maintenance mode."
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Ensures that the specified VMHost is restarted successfully in the specified period of time. If the elapsed time is
+    longer than the desired time for restart, the method throws an exception.
+    #>
+    [void] EnsureRestartTimeoutIsNotReached($elapsedTimeInSeconds) {
+        $timeSpan = New-TimeSpan -Seconds $elapsedTimeInSeconds
+        if ($this.RestartTimeout -le $timeSpan.Minutes) {
+            throw "Aborting the operation. VMHost $($this.Name) could not be restarted successfully in $($this.RestartTimeout) minutes."
+        }
+    }
+
+    <#
+    .DESCRIPTION
+
+    Ensures that the specified VMHost is in the desired state after successful restart operation.
+    #>
+    [void] EnsureVMHostIsInDesiredState($server, $desiredState) {
+        $sleepTimeInSeconds = 10
+        $elapsedTimeInSeconds = 0
+
+        while ($true) {
+            $this.EnsureRestartTimeoutIsNotReached($elapsedTimeInSeconds)
+
+            Start-Sleep -Seconds $sleepTimeInSeconds
+            $elapsedTimeInSeconds += $sleepTimeInSeconds
+
+            try {
+                if ($null -eq $server) {
```

Comment:
When $server is set to null? In case connection is direct to the ESX the connection will be dropped by the restart operation in those cases Connect-VIServer has to be called but I don't see how $server resets its value